subsys/ipc/ipc_service/ipc_service.c

```diff
@@ -81,3 +81,138 @@ int ipc_service_send(struct ipc_ept *ept, const void *data, size_t len)
 
 	return backend->send(ept->instance, ept->token, data, len);
 }
+
+int ipc_service_get_tx_buffer_size(struct ipc_ept *ept)
+{
+	const struct ipc_service_backend *backend;
+
+	if (!ept) {
+		LOG_ERR("Invalid endpoint");
+		return -EINVAL;
+	}
+
+	backend = ept->instance->api;
+
+	if (!backend || !backend->get_tx_buffer_size) {
+		LOG_ERR("Invalid backend configuration");
+		return -EIO;
+	}
+
+	return backend->get_tx_buffer_size(ept->instance, ept->token);
+}
+
+int ipc_service_get_tx_buffer(struct ipc_ept *ept, void **data, uint32_t *len, k_timeout_t wait)
+{
+	const struct ipc_service_backend *backend;
+
+	if (!ept || !data || !len) {
+		LOG_ERR("Invalid endpoint, data or len pointer");
+		return -EINVAL;
+	}
+
+	backend = ept->instance->api;
+
+	if (!backend || !backend->get_tx_buffer) {
+		LOG_ERR("Invalid backend configuration");
+		return -EIO;
+	}
+
+	if (!backend->send_nocopy) {
```

Comment:
test backend->get_tx_buffer here and adapt the message :
		LOG_ERR("No copy service not available");

Reply:
same comments can be applied on other "no copy" APIs

Reply:
This check is actually doing something slightly different. It's just making sure that we cannot have a `get_tx_buffer()` function without a corrispettive `send_nocopy()` function (and viceversa). A bit later we also check that we cannot have an `hold_rx_buffer()` function without a `release_rx_buffer()` call.

Reply:
Yes this is my point. What about following re-split?

```
if (!backend ) {
	/* The backend reference is not valid */
	return -EINVAL;
}
if (!backend->send_nocopy || !backend->get_tx_buffer) {
		LOG_ERR("No-copy service not available");
		return -EIO;
}
```
And same for `hold_rx_buffer()` & `release_rx_buffer()`